packages/build/src/plugins_core/deploy/buildbot_client.js

```diff
@@ -55,9 +56,16 @@ const getNextParsedResponsePromise = addAsyncErrorMessage(
   'Invalid response from buildbot',
 )
 
-const deploySiteWithBuildbotClient = async function (client, events, { PUBLISH_DIR }) {
+const deploySiteWithBuildbotClient = async function ({
+  client,
+  events,
+  buildDir,
+  repositoryRoot,
+  constants: { PUBLISH_DIR },
+}) {
   const action = shouldWaitForPostProcessing(events) ? 'deploySiteAndAwaitLive' : 'deploySite'
-  const payload = { action, deployDir: PUBLISH_DIR }
+  const deployDir = normalize(relative(repositoryRoot, resolve(buildDir, PUBLISH_DIR)))
```

Comment:
[dust] I find this hard a bit hard to parse, with three nested functions. If you can find an alternative way, or just adding a comment to explain what's going on, that might our future selves.

Reply:
Fixed :+1: